client-spring/src/main/java/com/netflix/conductor/client/spring/ConductorClientAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.client.spring;
+
+import com.netflix.conductor.client.automator.TaskRunnerConfigurer;
+import com.netflix.conductor.client.http.TaskClient;
+import com.netflix.conductor.client.worker.Worker;
+import com.netflix.discovery.EurekaClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties(ClientProperties.class)
+public class ConductorClientAutoConfiguration {
+
+    @Autowired(required = false)
+    private EurekaClient eurekaClient;
+
+    @Autowired
+    private List<Worker> workers;
```

Comment:
If there are no `Worker` instances, would the autowiring fail? Does this also need a `required=false` property?

Reply:
Yes, if no Worker instances bound, this would need to be optional.  I'm not sure what the client can accomplish without a worker though.

Reply:
Fair point.